plugins/utils/dressing.nix

```
@@ -0,0 +1,428 @@
+          Affects the dimensions of the window with respect to this setting.
+          If 'editor' or 'win', will default to being centered.
+        '';
+
+        prefer_width = helpers.defaultNullOpts.mkNullable intOrRatio "40" ''
+          Can be an integer or a float between 0 and 1 (e.g. 0.4 for 40%).
+        '';
+
+        width = helpers.defaultNullOpts.mkNullable intOrRatio "null" ''
+          Can be an integer or a float between 0 and 1 (e.g. 0.4 for 40%).
+        '';
+
+        max_width =
+          helpers.defaultNullOpts.mkNullable (with types; either intOrRatio (listOf intOrRatio))
+          "[140 0.9]"
+          ''
+            Max width of window.
+
+            Can be a list of mixed types, e.g. `[140 0.9]` means "less than 140 columns or 90% of
+            total."
+          '';
+
+        min_width =
+          helpers.defaultNullOpts.mkNullable (with types; either intOrRatio (listOf intOrRatio))
+          "[20 0.2]"
+          ''
+            Min width of window.
+
+            Can be a list of mixed types, e.g. `[140 0.9]` means "less than 140 columns or 90% of
+            total."
+          '';
+
+        buf_options = helpers.defaultNullOpts.mkAttrsOf types.anything "{}" ''
+          An attribute set of neovim buffer options.
+        '';
+
+        win_options =
+          helpers.defaultNullOpts.mkAttrsOf types.anything
+          ''
+            {
+              wrap = false;
+              list = true;
+              listchars = "precedes:...,extends:...";
+              sidescrolloff = 0;
+            }
+          ''
+          "An attribute set of window options.";
+
+        mappings =
+          helpers.defaultNullOpts.mkAttrsOf (with types; attrsOf (either str (enum [false])))
+          ''
+            {
+              n = {
+                "<Esc>" = "Close";
+                "<CR>" = "Confirm";
+              };
+              i = {
+                "<C-c>" = "Close";
+                "<CR>" = "Confirm";
+                "<Up>" = "HistoryPrev";
+                "<Down>" = "HistoryNext";
+              };
+            }
+          ''
+          ''
+            Mappings for defined modes.
+
+            To disable a default mapping in a specific mode, set it to `false`.
+          '';
+
+        override =
+          helpers.defaultNullOpts.mkLuaFn
+          ''
+            "function(conf) return conf end"
+          ''
+          ''
+            Lua function that takes config that is passed to nvim_open_win.
+            Used to customize the layout.
+          '';
+
+        get_config = helpers.defaultNullOpts.mkLuaFn "null" ''
+          This can be a function that accepts the opts parameter that is
+          passed in to 'vim.select' or 'vim.input'. It must return either
+          nil or config values to use in place of the global config values
+          for that module.
+
+          See :help dressing_get_config for more info.
+        '';
```

```suggestion
        get_config = helpers.defaultNullOpts.mkLuaFn "null" ''
          This can be a function that accepts the opts parameter that is
          passed in to 'vim.select' or 'vim.input'. It must return either
          nil or config values to use in place of the global config values
          for that module.

          See :help dressing_get_config for more info.
        '';
```

The line length limit is 100 characters. I guess that you can expand the lines a bit more here (and maybe elsewhere).

---

Done (also elsewhere)